src/ast.ts

```diff
@@ -705,6 +705,11 @@ export function decodeExpression(exp: Exp, outerPrecedence?: number): string {
         decodeExpression(expOp.args[1], innerPrecedence) + ' : ' +
         decodeExpression(expOp.args[2], innerPrecedence);
     }
+
+    if ((innerPrecedence >= outerPrecedence) && ((expOp.op === '&&') || (expOp.op === '||'))) {
+      result = '(' + result + ')';
+    }
```

Comment:
Can we move this to the `else if (expOp.args.length === 2)` clause?

Also, this comment from above, seems like it's supposed to handle this:

```
// All ops are left associative - so nudge the innerPrecendence
// down on the right hand side to force () for right-associating
// operations.
```

@tomlarkworthy are you familiar with any of this code?

Reply:
I noticed the comment and wondered about it, but as the tests show, it doesn't seem to be working as the comment describes.

I'll push the other change in a few hours.